src/mpc-hc/MainFrm.cpp

```diff
@@ -19878,3 +19872,142 @@ BOOL CMainFrame::AppendMenuEx(CMenu& menu, UINT nFlags, UINT_PTR nIDNewItem, CSt
     }
     return bResult;
 }
+
+CString CMainFrame::getBestTitle(bool fTitleBarTextTitle) {
+    CString title;
+    if (fTitleBarTextTitle) {
+        if (m_pAMMC) {
+            CComBSTR bstr;
+            if (SUCCEEDED(m_pAMMC->get_Title(&bstr)) && bstr.Length()) {
+                title = bstr.m_str;
+                title.Trim();
+                if (!title.IsEmpty()) {
+                    return title;
+                }
+            }
+        }
+    }
+
+    CPlaylistItem* pli = m_wndPlaylistBar.GetCur();
+    if (pli && !pli->m_fns.IsEmpty() && pli->m_label && !pli->m_label.IsEmpty()) {
+        if (fTitleBarTextTitle || pli->m_bYoutubeDL) {
+            title = pli->m_label;
+            return title;
+        }
+    }
+    return L"";
+}
+
+void CMainFrame::updateMediaTransControl() {
+    if (m_media_trans_control.updater) {
+        boolean enabled;
+        HRESULT ret = m_media_trans_control.controls->get_IsEnabled(&enabled);
+        ASSERT(ret == S_OK);
+        if (enabled) {

```

Comment:
The IF should also contain `ret == S_OK`.
The ASSERT only helps to detect unexpected results during debugging. We also need to handle unexpected situations in release builds.